lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -41,10 +43,17 @@ def fragment_cache(cached_hash, non_cached_hash)
 
         private
 
-        def add_relationships(resource, name, serializers)
+        def add_relationships(resource, name, serializers, opts)
           resource[:relationships] ||= {}
-          resource[:relationships][name] ||= { data: [] }
-          resource[:relationships][name][:data] += serializers.map { |serializer| { type: serializer.type, id: serializer.id.to_s } }
+
+          if opts.fetch(:data, true)
+            resource[:relationships][name] ||= { data: [] }
+            resource[:relationships][name][:data] += serializers.map { |serializer| { type: serializer.type, id: serializer.id.to_s } }
+          end
+          if opts.fetch(:links, false)
+            resource[:relationships][name] ||= {}
+            resource[:relationships][name][:links] = { related: url_helper.url_for([serializer.object, name]) }
```

Comment:
Did you consider using `Rails.application.routes.url_helpers` here instead of a reference to the controller?